### PR TITLE
MINOR: Update `TransactionalMessageCopier` to use the latest transaction pattern

### PR DESCRIPTION
**Pull request by dajac**
`ExactlyOnceMessageProcessor` is the reference so this patch updates `TransactionalMessageCopier` to be similar.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java [hunk 1]
```diff
@@ -361,7 +361,7 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
                             numMessagesProcessedSinceLastRebalance.getAndAdd(messagesSentWithinCurrentTxn);
                             totalMessageProcessed.getAndAdd(messagesSentWithinCurrentTxn);
                         }
-                    } catch (ProducerFencedException | OutOfOrderSequenceException e) {
+                    } catch (ProducerFencedException | FencedInstanceIdException e) {
```
Comment (hachikuji) [inline tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: nit: I don't think the copier uses group instance ID (maybe we could add support for that separately?), so I don't think `FencedInstanceIdException` is possible at the moment.

Comment (dajac) [reply tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: Indeed, it does not. Should we just remove `FencedInstanceIdException` for the time being?

Comment (hachikuji) [reply tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: Sounds good. I can file a jira to consider adding support. It's probably a worthwhile case to test.

Comment (dajac) [reply tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: Yeah, that's a good idea. We should likely add a new parameter to the test to choose if we want to enable this or not.

Comment (hachikuji) [reply tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: I filed this: https://issues.apache.org/jira/browse/KAFKA-13235.

Comment (dajac) [reply tools/src/main/java/org/apache/kafka/tools/TransactionalMessageCopier.java]: Removed the `FencedInstanceIdException` one.

**Review by hachikuji — APPROVED** [review]
LGTM

**Pull request merged**